llvm/lib/IR/RepoTicket.cpp

```diff
@@ -116,7 +116,37 @@ static GODigestState accumulateGOFinalDigestOrInitialDigestAndGetGODigestState(
   return accumulateGOInitialDigestAndGetGODigestState(GO, GOHash, GOIMap);
 }
 
-// Update the GO's hash value by adding the hash of its dependents.
+static void updateDigestUseContributions(const GlobalObject *GO, MD5 &GOHash,
+                                         const GOVec &Contributions,
+                                         GOInfoMap &GOIMap) {
+  for (const GlobalObject *const G : Contributions) {
+    assert(isa<GlobalVariable>(GO) &&
+           "Only global variables can have contributions!");
+    assert(isa<llvm::Function>(G) && "All contributions are functions!");
+    GOHash.update(GOIMap[G].InitialDigest.Bytes);
+  }
+}
+
+template <typename Function>
+static void updateDigestUseDependencies(MD5 &GOHash, const GOVec &Dependencies,
+                                        GOStateMap &Visited, GOInfoMap &GOIMap,
+                                        bool &Changed,
```

Comment:
Is there a reason why `Changed` is no returned from the function and instead passed by mutable reference?

Reply:
Thanks, Fixed.